aws/internal/service/iam/finder/finder.go

```diff
@@ -109,3 +111,22 @@ func Policies(conn *iam.IAM, arn, name, pathPrefix string) ([]*iam.Policy, error
 
 	return results, err
 }
+
+// RoleARNByName returns a role's ARN given the role name
+func RoleARNByName(conn *iam.IAM, name string) (*iam.Role, error) {
```

Comment:
Maybe rename to `RoleByName`? can be reused in other places as well.

Reply:
Cool, i renamed it

Reply:
I thought about adding retry to the finder. In the end I decided to not since we usually leave that for a waiter.